### PR TITLE
Scheduler/plugins/nodelabel fix no label preference

### PR DESCRIPTION
**Pull request by wzshiming**
#### What type of PR is this?
/kind bug
/sig scheduling

#### What this PR does / why we need it:

```
+++ [0309 17:15:15] Running tests without code coverage
E0309 17:15:42.866938   73156 runtime.go:78] Observed a panic: "integer divide by zero" (runtime error: integer divide by zero)
goroutine 5443 [running]:
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x4e212c0, 0x814f0d0)
        /Users/zsm/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0x95
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /Users/zsm/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x86
panic(0x4e212c0, 0x814f0d0)
        /Users/zsm/go/go1.16/src/runtime/panic.go:965 +0x1b9
k8s.io/kubernetes/pkg/scheduler/framework/plugins/nodelabel.(*NodeLabel).Score(0xc00675a7e0, 0x5f40d50, 0xc005c8ff00, 0xc007b6e990, 0xc00589f400, 0xc00708f334, 0xc, 0x1071b02, 0x5)
        /Users/zsm/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/scheduler/framework/plugins/nodelabel/node_label.go:130 +0x452
k8s.io/kubernetes/pkg/scheduler/framework/runtime.(*frameworkImpl).runScorePlugin(0xc00635f500, 0x5f40d50, 0xc005c8ff00, 0x31cb0340, 0xc00675a7e0, 0xc007b6e990, 0xc00589f400, 0xc00708f334, 0xc, 0x8, ...)
```
#### Which issue(s) this PR fixes:

xref #99937

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/nodelabel/node_label.go [hunk 1]
```diff
@@ -85,6 +85,11 @@ func (pl *NodeLabel) Name() string {
 // A node may have a label with "retiring" as key and the date as the value
 // and it may be desirable to avoid scheduling new pods on this node.
 func (pl *NodeLabel) Filter(ctx context.Context, _ *framework.CycleState, pod *v1.Pod, nodeInfo *framework.NodeInfo) *framework.Status {
+	size := int64(len(pl.args.PresentLabels) + len(pl.args.AbsentLabels))
+	if size == 0 {
+		return framework.NewStatus(framework.Skip)
```
[the remark this answers is not in the document]
Comment (ahg-g) [reply pkg/scheduler/framework/plugins/nodelabel/node_label.go:90]: right, just return nil here or revert the whole change.

**Comment by ahg-g** [comment]
relevant tests are failing

File: pkg/scheduler/framework/plugins/nodelabel/node_label.go [hunk 2]
```diff
@@ -85,6 +85,11 @@ func (pl *NodeLabel) Name() string {
 // A node may have a label with "retiring" as key and the date as the value
 // and it may be desirable to avoid scheduling new pods on this node.
 func (pl *NodeLabel) Filter(ctx context.Context, _ *framework.CycleState, pod *v1.Pod, nodeInfo *framework.NodeInfo) *framework.Status {
+	size := int64(len(pl.args.PresentLabels) + len(pl.args.AbsentLabels))
+	if size == 0 {
+		return nil
+	}
+
```
[the remark this answers is not in the document]
Comment (ahg-g) [reply pkg/scheduler/framework/plugins/nodelabel/node_label.go:92]: or keep it and update the test so that there is a label to check.

File: pkg/scheduler/factory.go [hunk 3]
```diff
@@ -200,7 +200,7 @@ func (c *Configurator) createFromProvider(providerName string) (*Scheduler, erro
 		prof := &c.profiles[i]
 		plugins := &schedulerapi.Plugins{}
 		plugins.Append(defaultPlugins)
-		plugins.Apply(prof.Plugins)
+		plugins.Append(prof.Plugins)
```
Comment (ahg-g) [inline pkg/scheduler/factory.go:203]: revert this change.

**Comment by wzshiming** [comment]
Sorry, I just pushed a wrong branch.

File: pkg/scheduler/factory.go (hunk 3 above)
Comment (wzshiming) [reply pkg/scheduler/factory.go]: Sorry, I just pushed a wrong branch.

**Comment by Huang-Wei** [comment]
Thanks @wzshiming! LGTM still.
